File: envy-data/src/main/java/absaliks/envy/util/FileUtil.java
package absaliks.envy.util;

import java.io.IOException;
import java.nio.charset.StandardCharsets;

public final class FileUtil {

  public static String readResource(String filename) {
    try (var stream = FileUtil.class.getResourceAsStream(filename)) {
      if (stream == null) {
        throw new RuntimeException("Resource not found: " + filename);
      }
      return new String(stream.readAllBytes(), StandardCharsets.UTF_8);
    } catch (IOException e) {
      throw new RuntimeException(e);
    }
  }
}

File: envy-data/src/main/kotlin/absaliks/envy/data/util/FileUtil.kt
package absaliks.envy.data.util

import java.io.File

object FileUtil {
  fun readResource(filename: String): String {
    val resource = this::class.java.getResource(filename) ?: throw RuntimeException("Resource not found: $filename")
    return resource.readText(Charsets.UTF_8)
  }

  fun readFile(filename: String): String {
    return File(filename).readText(Charsets.UTF_8)
  }
}